File: src/main/java/net/sharplab/epubtranslator/app/EPubTranslatorSetting.java
package net.sharplab.epubtranslator.app;


import org.eclipse.microprofile.config.inject.ConfigProperty;

import javax.enterprise.context.Dependent;

@Dependent
public class EPubTranslatorSetting {

    @ConfigProperty(name = "ePubTranslator.deepL.apiKey")
    String deepLApiKey;

    @ConfigProperty(name = "ePubTranslator.language.source")
    String defaultSrcLang;
    @ConfigProperty(name = "ePubTranslator.language.destination")
    String defaultDstLang;

    public String getDeepLApiKey() {
        return deepLApiKey;
    }

    public String getDefaultSrcLang() {
        return defaultSrcLang;
    }

    public String getDefaultDstLang() {
        return defaultDstLang;
    }
}

File: src/main/java/net/sharplab/epubtranslator/app/EPubTranslatorSetting.kt
package net.sharplab.epubtranslator.app

import org.eclipse.microprofile.config.inject.ConfigProperty
import jakarta.enterprise.context.Dependent

@Dependent
class EPubTranslatorSetting {
    @ConfigProperty(name = "ePubTranslator.deepL.apiEndpoint", defaultValue = "https://api.deepl.com")
    var deepLApiEndpoint: String? = null
    @ConfigProperty(name = "ePubTranslator.deepL.apiKey")
    var deepLApiKey: String? = null
    @ConfigProperty(name = "ePubTranslator.language.source", defaultValue = "en")
    var defaultSrcLang: String? = null
    @ConfigProperty(name = "ePubTranslator.language.destination", defaultValue = "ja")
    var defaultDstLang: String? = null

}
